src/lib.rs

```diff
@@ -177,6 +178,25 @@ impl<K: Hash + Eq, V, S: BuildHasher + Clone> WaitMap<K, V, S> {
     }
 }
 
+impl<K: Hash + Eq + fmt::Debug, V: fmt::Debug, S: BuildHasher + Clone> fmt::Debug for WaitMap<K, V, S> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut pmap = f.debug_map();
+        for r in self.map.iter() {
+            let (k, v) = r.pair();
+            pmap.entry(&k, &v);
+        }
+        pmap.finish()
+    }
+}
+
+impl<K: Hash + Eq, V, S: Default + BuildHasher + Clone> Default for WaitMap<K, V, S> {
+    #[inline]
+    fn default() -> Self {
+            Self::with_hasher(Default::default())
+    }
+}
+
+#[derive(Debug)]
```

Comment:
In the `Waiting` variant I think I'd want to see this print:

```
Waiting(waker count: N)
```

instead of:

```
Waiting([Waker, Waker, Waker, ..])
```

Otherwise :+1:!

Reply:
You got it, I'll rebase and fix it up.

Reply:
I did this with WakerSet's Debug impl, but I can move it to the enum if you'd like.